themes/default/content/blog/cloud-systems-part-two/index.md

```diff
@@ -0,0 +1,220 @@
+
+## Expanding the Website
+
+Our file structure gets more complicated now that we are building a more dynamic website with server-side routing and throwing it into a Docker container. You will have something like this:
+
+```
+container-tutorial
+├── Pulumi.yaml
+├── requirements.txt
+├── __main__.py
+└── website
+    ├── Dockerfile
+    ├── requirements.txt
+    └── app
+        ├── server.py
+        ├── static
+        │   ├── normalize.css
+        │   ├── style.css
+        │   └── background.jpg
+        └── templates
+            ├── base.html
+            ├── index.html
+            ├── portfolio.html
+            └── about.html
+```
+
+Everything contained within the `app` directory is the website itself. We now have server-side routing, and a few different pages. To run it locally and see what the new website looks like, `pip3 install flask` to install the Flask web framework and then run `python3 server.py` from the `app` directory. Flask will return an IP address where your website is running; navigate there in your browser and click around a bit!
+
+`server.py`
+
+```python
+from flask import Flask, render_template
+
+app = Flask(__name__)
+
+@app.route("/")
+def home():
+    return render_template("index.html")
+
+@app.route("/about")
+def about():
+    return render_template("about.html")
+
+@app.route('/portfolio')
+def portfolio():
+    return render_template("portfolio.html")
+
+
+if __name__ == '__main__':
+    app.run(host='0.0.0.0', port=80, debug=True)
+```
+
+Where before our entire website was made up of a single HTML file, we now have several, and we're using Flask to handle the routing between them. In this server file, three routs are defined: home, about, and portfolio. Flask handles serving these up for us, and associating a route with a particular HTML file. This way, we get a nice `www.mywebsite.com/about` URL instead of something like `www.mywebsite.com/about.html`. Building a website this way also means we have the ability to apply some logic to each of these routes, such as adding database interaction, user login, and passing conditional variables from the server to the templates that will be rendering each page. We aren't doing any of that yet, but we will!
+
+At the bottom, we're binding the Flask application to `0.0.0.0:80`. You can change that to any other unoccupied local IP address and port you like.
+
+## Templating
+
+Flask can make use of a templating engine called Jinja2 to make adding pages to your website a bit less repetitive. You start with one template, in this case called `base.html`, that contains any HTML you want to exist on every single page.
+
+`base.html`
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Hello world!</title>
```

Comment:
I'm guessing you're not changing the title on purpose for each page, right? Not a blocker; just curious.